### ISSUE TITLE
List workspace members

### ISSUE THREAD

**Issue by nicOwlas** [labelflow/labelflow#459]
## Tasks

In this first issue, we will create a view to display workspace users. In a second step, we will add the possibility to add new users.

[All mockups are here.](https://www.figma.com/file/4J9xyfF5v4B98MuKVO0Y8V/LabelFlow-mockups?node-id=2860%3A9381)

### Navigation bar

- [ ] Put the `Datasets` page in a tab
- [ ] Create a `Workspace Members` page and put it in a tab with the same name
- [ ] Adjust the breadcrumbs accordingly for `Datasets` and `Workspace Members`
- [ ] Set URL of the `Workspace Members` page to: `https://labelflow.ai/workspaceName/members`
- [ ] For the local workspace do not display the Workspace Members page

### Table of Members

The design is a copy-paste from [chakra table here](https://pro.chakra-ui.com/components/application-ui/tables). Some elements are removed. The table is described below.

- [ ] Title of the table: `Workspace Members (X)`. X: Number of Active + Invited members in the workspace
- [ ] Search bar: it filters the list of users which name or email address includes what is in the search bar
- [ ] New member button: it will be managed in another issue. Do not treat this in this issue.

#### 1st column: Profile picture

- [ ] Display the social login profile pic if the user logged in with Google / GitHub and the picture is available
- [ ] Fall back to [Chakra avatar](https://chakra-ui.com/docs/media-and-icons/avatar#avatar-fallbacks). With random color and the first letter of first name + first letter of the last name.

####  2nd column: Name

- [ ] Name of the user. If no name is available (typically when the user is being invited), display the username (the part before the `@`).
- [ ] Email address of the user

####  3rd column: Role

- [ ] Role. The person who created the workspace is the owner by default. The role is an unstyled chakra [select component](https://chakra-ui.com/docs/form/select#changing-the-appearance-of-the-select).
- [ ] Logic: Owner > Admin. A user can set any member to a role that is below or equal to his

####  4th column: Status

The status refers to the status of the membership as the user can belong to different workspaces.

- [ ] Add to the membership table a column `Status` which can be `Active` | `Invited` | `Deactivated`
- [ ] Ignore the `Resend invitation` in this issue. Do not display anything if a user is in the status `Invited`

#### 5th column: Activation

- [ ] A user can deactivate or reactivate a user having a role equal or inferior to his.
- [ ] Deactivated people cannot access the workspace anymore (note that they can access other online workspaces).
  - [ ] A user can deactivate himself (then present him this [modal](https://www.figma.com/file/4J9xyfF5v4B98MuKVO0Y8V/LabelFlow-mockups?node-id=2838%3A9425)) unless he is the sole owner of the workspace (then present him this [modal](https://www.figma.com/file/4J9xyfF5v4B98MuKVO0Y8V/LabelFlow-mockups?node-id=2838%3A9471)).
  - [ ] An owner can downgrade his role to an inferior role only if another owner is set else display this [modal](https://www.figma.com/file/4J9xyfF5v4B98MuKVO0Y8V/LabelFlow-mockups?node-id=2863%3A9439)

**Comment by etiennedupont** [comment]
Isn't this issue dependent on #351 @nicOwlas ?
I don't think this can be done without online workspaces

**Comment by nicOwlas** [comment]
> Isn't this issue dependent on #351 @nicOwlas ? I don't think this can be done without online workspaces

There is only this task that will be blocked by #296. I can create a specific issue for it. What point for #351 would be needed for this one?

**Comment by etiennedupont** [comment]
Yes indeed, it's rather #296. I think it might conflict a bit with this and maybe it'll make things easier to test.
Another question: what should be displayed on the member page when you're on `local/members` and you're not signed in?

**Comment by nicOwlas** [comment]
Good point. I updated the [mockup](https://www.figma.com/file/4J9xyfF5v4B98MuKVO0Y8V/LabelFlow-mockups?node-id=3050%3A9821) when non signed in:
![image](https://user-images.githubusercontent.com/16015322/135836742-3d88f475-d684-4b40-b5d8-5ea0f7647dcd.png)

**Comment by nacho-carnicero** [comment]
![](https://c.tenor.com/arrl5OtPFwIAAAAC/the-office-question.gif)

(Two actually):
 - Shouldn't we change the modal when the user is trying to remove him/herself? Right now it feels like he's deleting another person as the modal is the same
 - What behaviour should we do when a user removes him/herself from a workspace? Redirect to the local workspace?
